torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -716,7 +719,7 @@ void ProcessGroupNCCL::WorkNCCL::synchronizeInternal(
 // Same as calling synchronize().
 bool ProcessGroupNCCL::WorkNCCL::wait(std::chrono::milliseconds timeout) {
   RECORD_PARAM_COMMS(
-      static_cast<int>(this->seq_), // seq
+      std::make_tuple(static_cast<int64_t>(this->seq_), this->isP2P_), // seq
```

Comment:
Hmm I see. You put `isP2P_` into work because of this `RECORD_PARAM_COMMS` call in `work.wait()`.
Do you think we can infer isP2P from `work.opType`? (i.e. when it is `SEND` or `RECV`?). Or do you think there isn't a mapping at the moment?

Reply:
Would this example work for you?
https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L2472

Reply:
We can't because we can't deduce whether it's collective or p2p from `OpType::COALESCED` .

https://github.com/pytorch/pytorch/blob/a0207c8471989f13b0a30d7b532545793fc20cc1/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L2663